1 - Loading in count data, normalization & QC.Rmd

```diff
@@ -0,0 +1,195 @@
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+config <- yaml.load_file("config.yml")
+
+print(config$Paths$Treatment_vs_control_meta)
+```
+
+
+
+Load the CSV file with the read count data as well as the metadata file containing information about the sample grouping.
+The metadata file is needed for DESeq2's grouping.
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+# csv file must not be UTF-8 format
+# Read only if counts have been saved to CSV
+RNA_seq_data <- read.csv(data_file)
+
+# For Rh30
+# If there are duplicate gene names, take the sum of all read-counts.
+# TODO this can be improved upon.
+RNA_seq_counts_consolidated <- {RNA_seq_counts_df %>% group_by(Ensembl) %>% summarise_all(sum)}
+# Convert the gene symbols to the row names.
+RNA_seq_counts_final <- {RNA_seq_counts_consolidated %>% remove_rownames %>% column_to_rownames(var="Ensembl")}
+
+RNA_seq_meta_path <- config$Paths$Treatment_vs_control_meta
+RNA_seq_metadata <- read.table(RNA_seq_meta_path, header=T, row.names=1)
+```
+
+
+
+Check that sample names match in both files
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+# This must match for the DESeq2 object.
+all(colnames(RNA_seq_counts_final) %in% rownames(RNA_seq_metadata))
+all(colnames(RNA_seq_counts_final) == rownames(RNA_seq_metadata))
+```
+
+
+
+Plot RNA-seq count distribution for all samples
+```{r echo=FALSE, warning=FALSE, message=FALSE}
+# This can be done for other samples as well for QC-ing.
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Gene_xyz_KO_1), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Gene_xyz_KO_2), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Gene_xyz_KO_3), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Gene_xyz_KO_4), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Wildtype_1), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Wildtype_2), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Wildtype_3), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+
+ggplot(RNA_seq_counts_final) +
+  xlim(-5, 100) +
+  geom_histogram(aes(x = Wildtype_4), stat = "bin", bins = 100) +
+  xlab("Raw expression counts") +
+  ylab("Number of genes")
+```
+
+
+
+Modeling count data for Tumour spheres and Adherent conditions.
```

Comment:
For generalisation this should change to 'Modelling count data for Treatment and Control conditions'.

Reply:
ok